### PR TITLE
Improve cassette scrubbing/censoring

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Cassettes are now scrubbed recursively, so that nested JSON data that should be censored is addressed
- S/O to [`EasyVCR`](https://github.com/EasyPost/easyvcr-csharp/blob/a43acb8fdfb2de061407d59c035f0856bd5efef5/EasyVCR/Censors.cs#L240) for the implementation
- EndShipper unit tests switched to Test rather than Prod API key

# Testing

- All cassettes re-recorded and confirmed working on replay (censoring does not interfere)
- TestCustomsItemCreate failing on assertion, API returning `null` for item value?

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

File: tests/bootstrap_test.go [hunk 1]
```diff
@@ -41,29 +40,119 @@ type ClientTests struct {
...
+			list[index] = c.applyCensorsToJsonList(value.([]interface{}), elementsToCensor)
+		default:
+			// value is a single value or a normal list, nothing to censor
+		}
+	}
+
+	return list
+}
+
+func (c *ClientTests) applyCensorsToJsonDictionary(dictionary map[string]interface{}, elementsToCensor map[string]interface{}) map[string]interface{} {
```
Comment (Justintime50) [inline tests/bootstrap_test.go:80]: Missing docstring

**Review by Justintime50 — CHANGES_REQUESTED** [review]
1. Please add `fields` to the censor list since credential fields of a carrier get returned via that key
2. Please remove `children` as a key to censor since now it will recursively scrub the sensitive nested keys of that field while we can keep the rest of the data present.

File: tests/bootstrap_test.go [hunk 2]
```diff
@@ -41,29 +40,119 @@ type ClientTests struct {
...
+func contains(elements []string, target string) bool {
+	for _, element := range elements {
+		if target == element {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *ClientTests) applyCensorsToJsonList(list []interface{}, elementsToCensor map[string]interface{}) []interface{} {
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply tests/bootstrap_test.go:52]: Pretty self-explanatory, but resolved.

File: tests/bootstrap_test.go [hunk 3]
```diff
@@ -41,29 +40,119 @@ type ClientTests struct {
 	fixture  *Fixture
 }
 
+func contains(elements []string, target string) bool {
```
[the remark this answers is not in the document]
Comment (nwithan8) [reply tests/bootstrap_test.go:43]: Pretty self-explanatory, but resolved.

**Comment by nwithan8** [comment]
> 1. Please add `fields` to the censor list since credential fields of a carrier get returned via that key
> 2. Please remove `children` as a key to censor since now it will recursively scrub the sensitive nested keys of that field while we can keep the rest of the data present.

Do we need to update our other libraries? They are not censoring `fields`: https://github.com/EasyPost/easypost-csharp/blob/be20540414f493d3657f0cd63be086ee9de225e9/EasyPost.Tests/TestUtils.cs#L16

**Comment by Justintime50** [comment]
> Do we need to update our other libraries? They are not censoring `fields` and are censoring `children`: https://github.com/EasyPost/easypost-csharp/blob/be20540414f493d3657f0cd63be086ee9de225e9/EasyPost.Tests/TestUtils.cs#L16

This was fixed in PHP. We'll update the censored fields the same time we fix nested scrubbing. PHP/Go now, Python is in the works, the others to follow shortly but not to be handled separately.

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**